### PR TITLE
Square Pasture Java Solution

### PR DESCRIPTION
**Pull request by maggie-j-liu**
for cpi video classes

### CODE REVIEW

File: solutions/usaco-663.mdx [hunk 1]
```diff
@@ -24,20 +35,59 @@ int main()
...
 
 </CPPSection>
 
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class SquarePasture
```
Comment (SansPapyrus683) [inline solutions/usaco-663.mdx:64]:
```suggestion
public class Square
```

File: solutions/usaco-663.mdx [hunk 2]
```diff
@@ -24,20 +35,59 @@ int main()
...
+{
+	public static void main(String[] args) throws IOException
+	{
+		Kattio io = new Kattio("square");
+		int x1 = io.nextInt(), y1 = io.nextInt(), x2 = io.nextInt(), y2 = io.nextInt();
+		int x3 = io.nextInt(), y3 = io.nextInt(), x4 = io.nextInt(), y4 = io.nextInt();
+		// find the sides of the smallest rectangle covering both pastures
```
Comment (SansPapyrus683) [inline solutions/usaco-663.mdx:71]: maybe put each x and y pair on their own line

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-663.mdx (hunk 1 above)
Comment (maggie-j-liu) [reply solutions/usaco-663.mdx:64]: is there a reason for this? idk java soooo....

Comment (SansPapyrus683) [reply solutions/usaco-663.mdx]: so it matches the problem name

Comment (nathangong) [reply solutions/usaco-663.mdx]: Wait yeah I feel like SquarePasture would be better

Comment (nathangong) [reply solutions/usaco-663.mdx]: Oh nvm that's what Sans suggested. I think SquarePasture is a bit more descriptive of the problem.

Comment (maggie-j-liu) [reply solutions/usaco-663.mdx]: sans suggested changing SquarePasture -> Square

Comment (SansPapyrus683) [reply solutions/usaco-663.mdx]: if you're looking at the solution probably have an idea of what the problem is so

Comment (maggie-j-liu) [reply solutions/usaco-663.mdx]: > so it matches the problem name

the problem name is Square Pasture though...

Comment (nathangong) [reply solutions/usaco-663.mdx]: Oh ok I'm dumb yeah I think yours is better then

Comment (maggie-j-liu) [reply solutions/usaco-663.mdx]: i already changed it an i don't think it really matters, so i'll just leave it

**Review by SansPapyrus683 — APPROVED** [review]
nice

**Review by ryanchou-dev — APPROVED** [review]
looks good!

**Pull request merged**